stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/Message.java

```diff
@@ -315,32 +328,42 @@ public Message(
                         break;
                     }
                     case DescriptorProtos.FieldDescriptorProto.Type.TYPE_MESSAGE_VALUE: {
+                        BRecordType recordType = null;
+                        if (bType instanceof BRecordType) {
+                            recordType = (BRecordType) bType;
+                        }
+                        if (recordType == null) {
+                            throw Status.Code.INTERNAL.toStatus().withDescription("Error while decoding request " +
```

Comment:
Can't we have this logic inside the `else` block of the previous if block?

Reply:
Fixed in 6f7382cc88d67c386cfcc5e2d50f8c481c55aa06